modules/commerce_bluesnap_currency/src/EventSubscriber/ExchangeRateBluesnap.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace Drupal\commerce_bluesnap_currency\EventSubscriber;
+
+use Drupal\commerce_currency_resolver\ExchangeRateEventSubscriberBase;
+use Drupal\Component\Serialization\Json;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+
+/**
+ * Class ExchangeRateBluesnap.
+ *
+ * Fetches all currency conversion rates from bluesnap currency rate API.
+ */
+class ExchangeRateBluesnap extends ExchangeRateEventSubscriberBase {
+
+  /**
+   * The commerce currency resolver config.
+   *
+   * @var \Drupal\Core\Config\ImmutableConfig
+   */
+  protected $config;
+
+  /**
+   * Logger Factory.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelFactoryInterface
+   */
+  protected $loggerFactory;
+
+  /**
+   * Creates a new ExchangeRateBluesnap object.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The config factory.
+   * @param \Drupal\Core\Logger\LoggerChannelFactoryInterface $logger_factory
+   *   The logger factory.
+   */
+  public function __construct(
+    ConfigFactoryInterface $config_factory,
+    LoggerChannelFactoryInterface $logger_factory
+  ) {
+    $this->config = $config_factory->get('commerce_currency_resolver.currency_conversion');
+    $this->loggerFactory = $logger_factory->get('commerce_bluesnap_currency');
```

Comment:
This is not a logger factory, you are `get`ting the actual logger. Nevertheless see my comment above, please inject the existing logger. That will require enabling the main module; that's ok, I prefer that instead of creating another logger. It is extremely unlikely that somebody will fetch the BlueSnap exchange rates without using BlueSnap to receive payments.

We may also at some point switch to using our BlueSnap API client for fetching currency rates (do not change now).

Reply:
Used commerce bluesnap module logger.